src/zmalloc.c

```diff
@@ -101,10 +102,11 @@ static thread_local int thread_index = -1;
  * For the other architecture, lets fall back to the atomic operation to keep safe. */
 #if defined(__i386__) || defined(__x86_64__) || defined(__amd64__) || defined(__POWERPC__) || defined(__arm__) || \
     defined(__arm64__)
-static __attribute__((aligned(sizeof(size_t)))) size_t used_memory_thread[MAX_THREADS_NUM];
+static __attribute__((aligned(CACHE_LINE_SIZE))) size_t __used_memory_thread[MAX_THREADS_NUM + PADDING_ELEMENT_NUM];
```

Comment:
This padding seems wrong to me. Shouldn't we be padding the element instead? something like the below

```c
typedef struct {
    size_t value;
    char padding[CACHE_LINE_SIZE - sizeof(size_t)];  // Padding to ensure each element starts on a new cache line
} PaddedElement;

static __attribute__((aligned(CACHE_LINE_SIZE))) PaddedElement used_memory_thread[MAX_THREADS_NUM];
```

Reply:
Yeah, or align each element. Can we add the aligned attribute to the element struct?

Or does the aligned attribute on the array declaration actually apply to each element in the array?

Reply:
@PingXie @zuiderkwast Thanks for you proposal, I updated the top comment to add more explanation. 
The reason we didn't take the `struct padding` way is the performance degradation. Because when call function like `zmalloc_used_memory()` (always called in main thread), the main thread will cost more cycles to fetch more cache lines, per our benchmarking, it will have a ~3% performance degradation. 
Now we did some tradeoff, just fix the false sharing between main thread and io-threads, keep the false sharing in the io-threads for memory aggregation, because resource of main thread is bottleneck but resource of io-threads can be scaled.

Reply:
Thanks for the diagram, @lipzhu. That helps a lot. I agree this is a very reasonable trade-off. 

On a side note, I notice that the name of the new "private" array is prefixed with "__". I am fine with it but not sure if @zuiderkwast has any opinions :)

Reply:
@lipzhu thank for the explanation.

@PingXie Hehe, yeah, if we want to follow the C standard, any identifier starting with double underscore is reserved. The underscores are not descriptive either. I suggest add some descriptive word to the variable name, like `used_memory_thread_padded`.